code/modules/clothing/masks/gasmask.dm

```diff
@@ -2,25 +2,65 @@
 	name = "gas mask"
 	desc = "A face-covering mask that can be connected to an air supply. While good for concealing your identity, it isn't good for blocking gas flow." //More accurate
 	icon_state = "gas_alt"
-	clothing_flags = BLOCK_GAS_SMOKE_EFFECT | MASKINTERNALS
+	clothing_flags = BLOCK_GAS_SMOKE_EFFECT | MASKINTERNALS | GAS_FILTERING
 	flags_inv = HIDEEARS|HIDEEYES|HIDEFACE|HIDEFACIALHAIR|HIDESNOUT
 	w_class = WEIGHT_CLASS_NORMAL
 	inhand_icon_state = "gas_alt"
 	gas_transfer_coefficient = 0.01
 	permeability_coefficient = 0.01
 	flags_cover = MASKCOVERSEYES | MASKCOVERSMOUTH | PEPPERPROOF
 	resistance_flags = NONE
+	var/max_filters = 1
+	var/list/gas_filters
+
+/obj/item/clothing/mask/gas/examine(mob/user)
+	. = ..()
+	if(max_filters > 0)
+		. += "<span class='notice'>The amount of slots for filters are [max_filters].</span>"
+	if(LAZYLEN(gas_filters) > 0)
+		. += "<span class='notice'>Currently there are [LAZYLEN(gas_filters)] filters with [get_filter_durability()]% durability.</span>"
+
+/obj/item/clothing/mask/gas/attackby(obj/item/I, mob/user)
+	. = ..()
+	if(!istype(I, /obj/item/gas_filter))
+		return TRUE
+	if(LAZYLEN(gas_filters) >= max_filters)
+		return TRUE
+	if(!user.transferItemToLoc(I, src))
+		return TRUE
+	has_filter = TRUE
+	LAZYADD(gas_filters, I)
+
+/obj/item/clothing/mask/gas/consume_filter(datum/gas_mixture/breath)
+	if(LAZYLEN(gas_filters) <= 0 || max_filters == 0)
+		return breath
+	var/obj/item/gas_filter/gas_filter = pick(gas_filters)
+	var/datum/gas_mixture/filtered_breath = gas_filter.reduce_filter_status(breath)
+	if(gas_filter.filter_status <= 0)
+		has_filter = FALSE
+		LAZYREMOVE(gas_filters, gas_filter)
+		qdel(gas_filter)
+	return filtered_breath
+
+/obj/item/clothing/mask/gas/proc/get_filter_durability()
+	var/max_filters_durability = LAZYLEN(gas_filters) * 100
+	var/current_filters_durability
+	for(var/obj/item/gas_filter/gas_filter in gas_filters)
```

Comment:
```suggestion
	for(var/obj/item/gas_filter/gas_filter as anything in gas_filters)
```

Unless you, for whatever reason, expect `null` or non-gas filter related stuff to be in this list.